client/redux/user/index.js

```diff
@@ -0,0 +1,138 @@
+// import { loginUser, getUserData } from '../../api/user'
+import * as api from '../../api/user'
+import { getCookies } from '../../utils/getCookie'
+
+const LOG_OUT = 'LOG_OUT'
+const LOG_USER_REQUEST = 'LOG_USER_REQUEST'
+const LOG_USER_FAILURE = 'LOG_USER_FAILURE'
+const LOG_USER_SUCCESS = 'LOG_USER_SUCCESS'
+const SET_USER = 'SET_USER'
+
+export const logout = () => async (dispatch, getState) => {
+  const { userId } = getState().user
+  const token = getCookies('jwt')
+
+  try {
+    await api.logoutUser({ userId }, token)
+
+    dispatch({
+      type: LOG_OUT,
+    })
+  } catch (error) {
+    dispatch(logUserFailure(error))
+  }
+}
+
+export const logUserRequest = () => ({
+  type: LOG_USER_REQUEST,
+})
+
+export const logUserSuccess = (user) => ({
+  type: LOG_USER_SUCCESS,
+  payload: user,
+})
+
+export const logUserFailure = (error) => ({
+  type: LOG_USER_FAILURE,
+  payload: error,
+})
+
+const setUser = (id) => ({
+  type: SET_USER,
+  payload: id,
+})
+
+export const signIn = (user) => async (dispatch) => {
+  dispatch(logUserRequest(user))
+
+  try {
+    const { data } = await api.createUser(user)
+
+    const token = getCookies('jwt')
+
+    dispatch(setUser(data.user))
+    dispatch(getData(token))
+
+    document.location.replace('/')
+  } catch (error) {
+    const { message } = error.response.data
+
+    dispatch(logUserFailure(message))
+  }
+}
+
+export const login = (user) => async (dispatch) => {
+  dispatch(logUserRequest(user))
+
+  try {
+    const { data } = await api.loginUser(user)
+
+    const token = getCookies('jwt')
+
+    dispatch(setUser(data.user))
+    dispatch(getData(token))
+
+    document.location.replace('/')
+  } catch (error) {
+    const { message } = error.response.data
+
+    dispatch(logUserFailure(message))
+  }
+}
+
+export const getData = (token) => async (dispatch) => {
+  try {
+    const { data } = await api.getUserData(token)
+
+    dispatch(logUserSuccess(data))
+  } catch (error) {
+    dispatch(logUserFailure(error))
+  }
+}
+
+const initialState = {
+  currentUser: {},
+  isAuth: getCookies('jwt') ? getCookies('jwt') : false,
```

Comment:
@AStashevskaya Try to avoid executing function with same input params twice, where you can execute it once and save result

```js
const jwtCookie = getCookies('jwt')

const initialState ={
  currerntUser: {},
  isAuth: jwtCookie ? jwtCookie : false,
  userId: '',
  loading: false,
  error: '',
}
```